components/nodemanager-service/nodemanager.go

```diff
@@ -103,23 +134,121 @@ func serve(ctx context.Context, config *config.Nodemanager, connFactory *securec
 	if eventClient == nil {
 		return fmt.Errorf("nodemanager setup, could not obtain automate events service client")
 	}
-	// start polling for node mgrs, passing along the secrets client and event client
-	// this function will call each of the manager polling functions that check the
-	// state of nodes at the configured intervals
-	polling.StartThePolls(ctx, config.Manager, db, secretsClient, eventClient)
 
-	// create Automate node manager to handle manually added nodes. all manually created
-	// nodes are assigned to the Automate node manager
-	log.Debug("creating Automate node manager")
+	// Set the cereal service client
+	log.Infof("nodemanager setup, getting a cereal service client %s", config.Cereal.Endpoint)
+	cerealConn, err := connFactory.DialContext(timeoutCtx, "cereal-service", config.Cereal.Endpoint, grpc.WithBlock())
+	if err != nil {
+		err = errors.New("nodemanager setup, error grpc dialing to cereal-service aborting...")
+		return err
+	}
+	// LATER: look at whatever Dan did for picking the domain
+	cerealBackend := grpccereal.NewGrpcBackendFromConn("nodemanager", cerealConn)
+
+	// Set up a cereal (workflow) manager for managing nodemanager workflows.
+	cerealManager, err := cereal.NewManager(cerealBackend)
+	if err != nil {
+		return errors.Wrap(err, "could not create cereal manager")
+	}
+	defer cerealManager.Stop() //nolint:errcheck
+
+	// Prelude to initializing the job manager: register task executors for the nodemanger polling jobs
+	err = cerealManager.RegisterTaskExecutor(AwsEc2PollingJobName,
+		&CheckAWSNodesTask{db: db, secretsClient: secretsClient, eventsClient: eventClient},
+		cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+
+	err = cerealManager.RegisterTaskExecutor(AzureVMPollingJobName,
+		&CheckAzureNodesTask{db: db, secretsClient: secretsClient}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+
+	err = cerealManager.RegisterTaskExecutor(ManagersPollingJobName,
+		&CheckManagersTask{db: db, secretsClient: secretsClient}, cereal.TaskExecutorOpts{})
+	if err != nil {
+		return err
+	}
+
+	// Prelude to initializing the cereal manager: register workflow executors to run tasks
+	for _, jobName := range []string{AwsEc2PollingJobName, AzureVMPollingJobName, ManagersPollingJobName} {
+		err = cerealManager.RegisterWorkflowExecutor(jobName, patterns.NewSingleTaskWorkflowExecutor(jobName, false))
+		if err != nil {
+			return errors.Wrapf(err, "failed to register workflow for %q", jobName)
+		}
+	}
+
+	kindsOfChecks := [3]string{"aws", "azure", "manager"}
+	var pollInterval int
```

Comment:
do these need to be declared outside the for loop?

Reply:
Nope -- I'll move them.